src/PVRIptvData.cpp

```diff
@@ -1482,3 +1487,26 @@ int PVRIptvData::GetChannelId(const char * strChannelName, const char * strStrea
 
   return abs(iId);
 }
+
+std::string& PVRIptvData::AddHeaderToStreamUrl(std::string& strLine, const std::string& header)
+{
+  size_t found = strLine.find('|');
+
+  if (found != std::string::npos)
+  {
+    size_t present = strLine.find(header + '=', found + 1);
+
+    if (present == std::string::npos)
+    {
+      strLine += '&';
+    }
+
+    return strLine;
+  }
+  else
+  {
+    strLine += '|';
+  }
+
+  return strLine += header + '=' + g_userAgent;
```

Comment:
This statement could just be in the else above as it will only be executed if the else is.

Reply:
This statement must be executed in two cases, so instead of putting in both, i decided to write once at the end.

Reply:
Makes sense, although modifying the strLine variable and then returning the result doesn't make sense. I think it will confuse other devs. Maybe just return the value.

Reply:
That flow never happens. That return it's inside an else indicating the header was already present, so the playlist has priority over user setting.

Reply:
But why does this make sense:
`return strLine += header + '=' + g_userAgent;`

Why not just just return `strLine + header + '=' + g_userAgent;'

Reply:
It's not a prob, used this one because it was committed by you on the Matrix branch.